File: app/src/main/java/me/mitul/aij/adapter/AdapterViewPager.java
package me.mitul.aij.adapter;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import java.util.ArrayList;

public class AdapterViewPager extends FragmentPagerAdapter {
    private final ArrayList<String> titleList = new ArrayList<>();
    private final ArrayList<Fragment> fragmentList = new ArrayList<>();

    public AdapterViewPager(FragmentManager manager) {
        super(manager);
    }

    public void addFragment(Fragment fragment, String title) {
        fragmentList.add(fragment);
        titleList.add(title);
    }

    public int getCount() {
        return fragmentList.size();
    }

    @NonNull
    public Fragment getItem(int index) {
        return fragmentList.get(index);
    }

    public CharSequence getPageTitle(int index) {
        return titleList.get(index);
    }
}

File: app/src/main/java/me/mitul/aij/adapter/AdapterViewPager.kt
package me.mitul.aij.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

class AdapterViewPager(
    manager: FragmentManager,
    private val list: ArrayList<String> = ArrayList(),
    private val fragments: ArrayList<Fragment> = ArrayList(),
) : FragmentPagerAdapter(manager) {
    override fun getCount() = fragments.size

    override fun getItem(index: Int) = fragments[index]

    override fun getPageTitle(index: Int) = list[index]

    fun add(title: String, fragment: Fragment) {
        list.add(title)
        fragments.add(fragment)
    }
}
